draco/analysis/dayenu.py

```diff
@@ -615,18 +615,38 @@ def process(self, hv, source):
         # loop over products
         for tt in range(ntime):
             t0 = time.time()
+            self.log.debug(f"Filter time {tt} of {ntime}.")
 
             for xx in range(new):
-                self.log.debug(f"Filter time {tt} of {ntime}, baseline {xx} of {new}.")
 
                 for pp in range(npol):
+
+                    flag = weight[pp, :, xx, tt] > 0.0
+
+                    # Skip full masked samples
+                    if not np.any(flag):
+                        continue
+
                     # Grab datasets for this pol and ew baseline
                     tvis = np.ascontiguousarray(vis[pp, :, xx, :, tt])
                     tvar = tools.invert_no_zero(weight[pp, :, xx, tt])
 
                     # Grab the filter for this pol and ew baseline
                     NF = np.ascontiguousarray(filt[pp, :, :, xx, tt])
 
+                    # Make sure that any frequencies unmasked during filter generation
+                    # are also unmasked in the data
+                    valid_freq = np.any(np.abs(NF) > 0.0, axis=0)
+                    missing_freq = np.flatnonzero(valid_freq & ~flag)
+                    if missing_freq.size > 0:
+                        self.log.warning(
+                            "Missing the following frequencies that were "
+                            "assumed valid during filter generation: "
+                            f"{missing_freq}"
+                        )
+                        weight[pp, :, xx, tt] = 0.0
```

Comment:
don't you want, the missing freq would be 0 in the weight?

Reply:
If there are missing frequencies, then applying the filter won't work at all, since those frequencies were not taken into account during filter generation.  This will just set the weight to zero for all frequencies if that happens.